src/mame/konami/kpython2.cpp

```diff
@@ -0,0 +1,1346 @@
+- 1 attachment device
+Port 1, node 1: ICCA card reader
+Port 2: Unpopulated
+
+Thrill Drive:
+Will give a data warning on first boot after installation. Press test button to skip.
+
+- 2 attachment devices
+Port 1: Unpopulated?
+Port 2, node 1: Steering wheel device ("HNDL")
+Port 2, node 2: Seat belt device ("BELT")
+
+Toy's March:
+- 1 attachment device
+Port 1: Drum pad I/O
+Port 2: Unpopulated
+
+Dance 86.4:
+Uses the same GN845-PWB(B) "DDR Stage Multiplexor" as defined in ksys573.cpp.
+The output data bit and clock are sent through the lamp/output command (0x24)
+through endpoint 1 on the USB device. The 0x24 command functions in the same
+way as ksys573's ddr_output_callback.
+
+
+
+TODO:
+    Python 2 is based on consumer PS2, should be derived from ps2sony.cpp
+    Everything
+
+***************************************************************************/
+
+
+#include "emu.h"
+#include "cpu/mips/mips3.h"
+#include "cpu/mips/mips1.h"
+//#include "machine/ds2430.h"
+#include "machine/timekpr.h"
+#include "emupal.h"
+#include "screen.h"
+
+
+class kpython2_state : public driver_device
+{
+public:
+	kpython2_state(const machine_config &mconfig, device_type type, const char *tag)
+		: driver_device(mconfig, type, tag)
+		, m_maincpu(*this, "maincpu")
+		, m_ps2_nvram(*this, "ps2_nvram")
+		, m_ps2_hdd_id(*this, "ps2_hdd_id")
+		, m_ds2430_black_rom(*this, "ds2430_black")
+		, m_ds2430_white_rom(*this, "ds2430_white")
+	{ }
+
+	void kpython2(machine_config &config);
+
+private:
+	uint32_t screen_update(screen_device &screen, bitmap_rgb32 &bitmap, const rectangle &cliprect);
+
+	void ps2_map(address_map &map);
+
+	// devices
+	required_device<mips3_device> m_maincpu;
+
+	required_region_ptr<uint8_t> m_ps2_nvram; // Contains an ILINK ID that must match the one encrypted on the HDD
+	required_region_ptr<uint8_t> m_ps2_hdd_id; // Must match the HDD that the data was encrypted against
+
+	required_region_ptr<uint8_t> m_ds2430_black_rom;
+	required_region_ptr<uint8_t> m_ds2430_white_rom;
+
+	// driver_device overrides
+	virtual void video_start() override;
+};
+
+
+void kpython2_state::video_start()
+{
+}
+
+uint32_t kpython2_state::screen_update(screen_device &screen, bitmap_rgb32 &bitmap, const rectangle &cliprect)
+{
+	return 0;
+}
+
+void kpython2_state::ps2_map(address_map &map)
+{
+	map(0x00000000, 0x01ffffff).ram();
+	map(0x1fc00000, 0x1fdfffff).rom().region("bios", 0);
+}
+
+void kpython2_state::kpython2(machine_config &config)
+{
+	R5000LE(config, m_maincpu, 294000000); // imported from namcops2.c driver
+	m_maincpu->set_icache_size(16384);
+	m_maincpu->set_dcache_size(16384);
+	m_maincpu->set_addrmap(AS_PROGRAM, &kpython2_state::ps2_map);
+
+	// Video hardware
+	screen_device &screen(SCREEN(config, "screen", SCREEN_TYPE_RASTER));
+	screen.set_refresh_hz(60);
+	screen.set_screen_update(FUNC(kpython2_state::screen_update));
+	screen.set_size(640, 480);
+	screen.set_visarea(0, 639, 0, 479);
+
+	PALETTE(config, "palette").set_entries(65536);
```

Comment:
You might want to use the devices from src/mame/sony/ps2sony.cpp as a starting point – it’s closer to being a real PlayStation 2 driver than anything else AFAIK.

Reply:
That's fair. I copypasted kpython.cpp as the base of this and made some small changes to remove the irrelevant code. I've redone it to use ps2sony.cpp as the base.